packages/world/ts/actions/callFrom.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  slice,
+  concat,
+  type WalletClient,
+  type Transport,
+  type Chain,
+  type Account,
+  type Hex,
+  type WalletActions,
+  type WriteContractReturnType,
+  type EncodeFunctionDataParameters,
+} from "viem";
+import { getAction, encodeFunctionData } from "viem/utils";
+import { writeContract } from "viem/actions";
+
+type CallFromParameters = {
+  worldAddress: Hex;
+  delegatorAddress: Hex;
+  worldFunctionToSystemFunction: (
+    worldFunctionSelector: Hex,
+  ) => Promise<{ systemId: Hex; systemFunctionSelector: Hex }>;
+};
+
+// By extending viem clients with this function after delegation, the delegation is automatically
+// applied to the World contract writes, meaning these writes are made on behalf of the delegator.
+export function callFrom<TChain extends Chain, TAccount extends Account>({
+  worldAddress,
+  delegatorAddress,
+  worldFunctionToSystemFunction,
```

Comment:
thought about this some more and since we already assume you're operating on a world and function selectors never change once set (i.e. we can safely cache them here), I think we can get rid of this option in favor of something like:

```ts
type SystemFunction = { readonly systemId: Hex, readonly systemFunctionSelector: Hex };
const worldFunctionSelectorCache = new Map<Hex, SystemFunction>();
async function worldFunctionToSystemFunction(client: PublicClient, worldAddress: Address, worldFunctionSelector: Hex): Promise<SystemFunction> {
  const cached = worldFunctionSelectorCache.get(worldFunctionSelector);
  if (cached != null) return cached;

  const systemFunction = await readContract(client, {
    address: worldAddress,
    abi: worldAbi,
    functionName: 'getRecord',
    args: [worldConfig.tables.FunctionSelectors.tableId, worldFunctionSelector]
  });
  worldFunctionSelectorCache.set(worldFunctionSelector, systemFunction);
  return systemFunction;
}
```
(Note that this is pseudocode and may require parsing the record from the `getRecord` call, and I'm not sure how best to expect the public client, wether we expect to extend a public client or take it as an argument)

Reply:
Right, it's better to have that automatic retrieving, so users without this info sync can use this function more easily.

Getting a public client as an argument is more flexible, as it doesn't ask users to extend their wallet clients with public actions, so they have options. I'll implement this.